src/MOIWrapper.jl

```diff
@@ -0,0 +1,345 @@
+    f::MOI.SingleVariable, s::MOI.EqualTo)
+    cbcModelFormat.col_lb[mapping.varmap[f.variable].value] = s.value
+    cbcModelFormat.col_ub[mapping.varmap[f.variable].value] = s.value
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.SingleVariable, s::MOI.LessThan)
+    cbcModelFormat.col_ub[mapping.varmap[f.variable].value] = s.upper
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.SingleVariable, s::MOI.GreaterThan)
+    cbcModelFormat.col_lb[mapping.varmap[f.variable].value] = s.lower
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.SingleVariable, s::MOI.Interval)
+    cbcModelFormat.col_lb[mapping.varmap[f.variable].value] = s.lower
+    cbcModelFormat.col_ub[mapping.varmap[f.variable].value] = s.upper
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat,
+    mapping::MOIU.IndexMap, f::MOI.ScalarAffineFunction, s::MOI.EqualTo)
+    for term in f.terms
+        cbcModelFormat.constraint_matrix[mapping.conmap[ci].value,
+        mapping.varmap[term.variable_index].value] += term.coefficient
+    end
+    cbcModelFormat.row_lb[mapping.conmap[ci].value] = s.value - f.constant
+    cbcModelFormat.row_ub[mapping.conmap[ci].value] = s.value - f.constant
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat,
+    mapping::MOIU.IndexMap, f::MOI.ScalarAffineFunction, s::MOI.GreaterThan)
+    for term in f.terms
+        cbcModelFormat.constraint_matrix[mapping.conmap[ci].value,
+        mapping.varmap[term.variable_index].value] += term.coefficient
+    end
+    cbcModelFormat.row_lb[mapping.conmap[ci].value] = s.lower - f.constant
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.ScalarAffineFunction, s::MOI.LessThan)
+    for term in f.terms
+        cbcModelFormat.constraint_matrix[mapping.conmap[ci].value,
+        mapping.varmap[term.variable_index].value] += term.coefficient
+    end
+    cbcModelFormat.row_ub[mapping.conmap[ci].value] = s.upper - f.constant
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.ScalarAffineFunction, s::MOI.Interval)
+    for term in f.terms
+        cbcModelFormat.constraint_matrix[mapping.conmap[ci].value,
+        mapping.varmap[term.variable_index].value] += term.coefficient
+    end
+    cbcModelFormat.row_ub[mapping.conmap[ci].value] = s.upper - f.constant
+    cbcModelFormat.row_lb[mapping.conmap[ci].value] = s.lower - f.constant
+end
+
+
+function loadObj(cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.ScalarAffineFunction)
+    for term in f.terms
+        cbcModelFormat.obj[mapping.varmap[term.variable_index].value] += term.coefficient
```

Comment:
This assumes that `obj` is all zeros at the start, right? Add a comment mentioning this assumption.

Reply:
When cbc_model_format is created everything is initialized to 0.0. So I do `+=` to handle cases where the user sets an objective function like $x_1 + x_2 + x_1$. These cases are tested in `contlinear.jl`.

Reply:
That's all good, but please add a comment about this assumption right here.